lib/multicell_location/services/here_integration.c

```diff
@@ -70,24 +80,96 @@ BUILD_ASSERT(sizeof(API_APP_ID) > 1, "App ID must be configured");
 	"{"								\
 		"\"lte\":["						\
 			"{"						\
-				"\"mcc\": %d,"				\
-				"\"mnc\": %d,"				\
-				"\"cid\": %d"				\
+				"\"mcc\":%d,"				\
+				"\"mnc\":%d,"				\
+				"\"cid\":%d,"				\
+				"\"tac\":%d,"				\
+				"\"rsrp\":%d,"				\
+				"\"rsrq\":%.1f,"			\
+				"\"ta\":%d"				\
 			"}"						\
 		"]"							\
 	"}"
 
 #define HTTP_REQUEST_BODY_NEIGHBOR_ELEMENT				\
 	"{"								\
-		"\"earfcn\": %d,"						\
-		"\"pci\": %d"						\
+		"\"earfcn\":%d,"					\
+		"\"pci\":%d,"						\
+		"\"rsrp\":%d,"						\
+		"\"rsrq\":%.1f"						\
 	"}"
 
 BUILD_ASSERT(sizeof(HOSTNAME) > 1, "Hostname must be configured");
 
 static char body[HTTP_BODY_SIZE];
 static char neighbors[NEIGHBOR_BUFFER_SIZE];
 
+/* Modem returns RSRP and RSRQ as index values which require
+ * a conversion to dBm and dB respectively. See modem AT
+ * command reference guide for more information.
+ */
+#define RSRP_ADJ(rsrp) (rsrp - ((rsrp <= 0) ? 140 : 141))
```

Comment:
Yes, I've been wondering that formula in AT command spec, They are saying:
95 – When −46 dBm ≤ RSRP < −45 dBm

The index value of RSRP can be converted to decibel with the following formula:
Index – 140 = dBm. For example, 95 – 140 = –45 dBm

=> -45 is not in range. it is not < -45 but when calculating with 141 then it is in range = -46.

This same formula seems to be used elsewhere in cellular location, thus maybe better to keep this as it is to be aligned.

Reply:
Yes, this is a bit confusing. Let's keep it like this for now, but also it would be good to confirm from the modem team how the mapping should actually be done. Looks like the description in the AT command spec should be clarified.